shipyard/src/main/java/org/jclouds/shipyard/handlers/ShipyardErrorHandler.java

```diff
@@ -90,7 +94,7 @@ private String parseMessage(HttpResponse response) {
       try {
          return Strings2.toStringAndClose(response.getPayload().openStream());         
       } catch (IOException e) {
-         throw Throwables.propagate(e);
+         return Throwables.getStackTraceAsString(e);
```

Comment:
I'd revert this change. If you do this, users may get an HttpResponseException that means that the target provider returned an error response but instead of having the real error message you will be silently return a complete different error: the error jclouds found when trying to read the error stream.

Let's fail if you can't read it and propagate that unexpected exception.